client/wfnews-war/src/main/angular/src/background.js

```diff
@@ -0,0 +1,111 @@
+const Storage = require('@ionic/storage');
+
+function submitOfflineRoF ( event ) {
+  addEventListener(event, (resolve, reject) => { 
+    try {
+        scheduleDataSync();
+        resolve();
+    } catch (err) {
+        reject(err);
+    }
+  });
+}
+
+ async function scheduleDataSync() {
+    const storage = new Storage();
+    await storage.create();
+    const isConnected = await checkOnlineStatus();
+      if (isConnected) {
+        await syncDataWithServer();
+      }
+    }
+
+ async function syncDataWithServer() {
+  const storage = new Storage();
+    try {
+      // Fetch and submit locally stored data
+      const offlineReport = await storage.get('offlineReportData');
+
+      if (offlineReport) {
+        // Send the report to the server
+        const response = await submitReportToServer(offlineReport);
+
+        if (response.success) {
+          // Remove the locally stored data if sync is successful
+          await storage.remove('offlineReportData');
+        }
+      }
+    } catch (error) {
+      console.error('Sync failed:', error);
+    }
+  }
+
+async function submitReportToServer(offlineReport){
+    const rofJson = JSON.parse(offlineReport);
+    const resource = rofJson.resource;
+    const image1 = rofJson.image1;
+    const image2 = rofJson.image2;
+    const image3 = rofJson.image3;
+    const rofUrl = "https://wfone-notifications-api.dev.bcwildfireservices.com/rof"
```

Comment:
No hard-coded URL's, use an environment specific parameter. This should be in our secrets and applied on build, fetched from the app properties